packages/regenesis-surgery/test/predeploy.spec.ts

```diff
@@ -1,56 +1,178 @@
-/* eslint-disable @typescript-eslint/no-empty-function */
 import { expect, env } from './setup'
-import { AccountType, Account } from '../scripts/types'
+import { AccountType } from '../scripts/types'
+import { GenesisJsonProvider } from './provider'
 
 describe('predeploys', () => {
-  let predeploys = {
+  const predeploys = {
     eth: [],
     newNotEth: [],
     noWipe: [],
     wipe: [],
     weth: [],
   }
+  // Base genesis file only
+  let genesisStateProvider: GenesisJsonProvider
+  // Old sequencer state
+  let oldStateProvider: GenesisJsonProvider
 
   before(async () => {
     await env.init()
     predeploys.eth = env.getAccountsByType(AccountType.PREDEPLOY_ETH)
-    predeploys.newNotEth = env.getAccountsByType(AccountType.PREDEPLOY_NEW_NOT_ETH)
+    predeploys.newNotEth = env.getAccountsByType(
+      AccountType.PREDEPLOY_NEW_NOT_ETH
+    )
     predeploys.noWipe = env.getAccountsByType(AccountType.PREDEPLOY_NO_WIPE)
     predeploys.wipe = env.getAccountsByType(AccountType.PREDEPLOY_WIPE)
     predeploys.weth = env.getAccountsByType(AccountType.PREDEPLOY_WETH)
+
+    genesisStateProvider = new GenesisJsonProvider(
```

Comment:
The providers can be found on the `env`, see https://github.com/ethereum-optimism/optimism/blob/e4ada1dde87055bf68ed97ee6af4171b487cd4d2/packages/regenesis-surgery/test/setup.ts#L59

Reply:
the `preL2Provider` has the genesis and state dump merged, whereas some of these tests make a distinction between base genesis and pre-surgery state, which is why i initialized the new providers